src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -154,28 +154,48 @@ private void traverseArray(ArrayNode arr, ShredListener callback, JsonPath.Build
 
   private void traverseValue(JsonNode value, ShredListener callback, JsonPath.Builder pathBuilder) {
     final JsonPath path = pathBuilder.build();
-    if (value.isObject()) {
-      ObjectNode ob = (ObjectNode) value;
-      if (callback.shredObject(path, ob)) {
-        traverseObject(ob, callback, pathBuilder.nestedObjectBuilder());
+    if (path.toString().equals(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD)) {
+      if (value.isNull()) {
+        return;
+      }
+      if (!value.isArray()) {
+        throw new JsonApiException(
+            ErrorCode.SHRED_BAD_DOCUMENT_VECTOR_TYPE,
+            String.format(
+                "%s: %s",
+                ErrorCode.SHRED_BAD_DOCUMENT_VECTOR_TYPE.getMessage(), value.getNodeType()));
       }
-    } else if (value.isArray()) {
       ArrayNode arr = (ArrayNode) value;
-      callback.shredArray(path, arr);
-      traverseArray(arr, callback, pathBuilder.nestedArrayBuilder());
-    } else if (value.isTextual()) {
-      callback.shredText(path, value.textValue());
-    } else if (value.isNumber()) {
-      callback.shredNumber(path, value.decimalValue());
-    } else if (value.isBoolean()) {
-      callback.shredBoolean(path, value.booleanValue());
-    } else if (value.isNull()) {
-      callback.shredNull(path);
+      if (arr.size() == 0) {
```

Comment:
Should we also confirm that the vector has the correct length?

Reply:
This case will depend on the cassandra error. I will add an integration test for this.

Reply:
Ah, because schema information we have tells about tables, column names, but has no fine-grained column info, needed to determine size.

Reply:
@vkarpov15 The base cassandra image need fix for this. Will do different PR for this case.